core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -3472,6 +3483,312 @@ protected UniqueIDPaginatedSearchResult doGetUserListWithID(Condition condition,
+        return getGroupByGroupName(groupName, null);
+    }
+
+    protected void persistGroup(String groupName, String groupId, List<String> userIds) throws UserStoreException {
+
+        Connection dbConnection = null;
+        String sqlStmt;
+
+        try {
+            dbConnection = getDBConnection();
+            sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.ADD_GROUP);
+            if (sqlStmt.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                this.updateStringValuesToDatabase(dbConnection, sqlStmt, groupId, groupName, tenantId, new Date(),
+                        new Date());
+            } else {
+                this.updateStringValuesToDatabase(dbConnection, sqlStmt, groupId, groupName, new Date(), new Date());
+            }
+
+            if (userIds != null) {
+                String[] userIdList = userIds.toArray(new String[0]);
+                // add group to user
+                String type = DatabaseCreator.getDatabaseType(dbConnection);
+                String sqlStmt2 = realmConfig.getUserStoreProperty(JDBCRealmConstants.ADD_USER_TO_ROLE_WITH_ID + "-" + type);
+
+                if (sqlStmt2 == null) {
+                    sqlStmt2 = realmConfig.getUserStoreProperty(JDBCRealmConstants.ADD_USER_TO_ROLE_WITH_ID);
+                }
+                if (sqlStmt2.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                    if (UserCoreConstants.OPENEDGE_TYPE.equals(type)) {
+                        DatabaseUtil.udpateUserRoleMappingInBatchMode(dbConnection, sqlStmt2, tenantId, userIdList,
+                                tenantId, groupName, tenantId);
+                    } else {
+                        DatabaseUtil.udpateUserRoleMappingInBatchMode(dbConnection, sqlStmt2, userIdList, tenantId,
+                                groupName, tenantId, tenantId);
+                    }
+                } else {
+                    DatabaseUtil.udpateUserRoleMappingInBatchMode(dbConnection, sqlStmt2, userIdList, groupName);
+                    DatabaseUtil.udpateUserRoleMappingInBatchMode(dbConnection, sqlStmt2, userIdList, groupName);
+                }
+            }
+            dbConnection.commit();
+        } catch (SQLException e) {
+            DatabaseUtil.rollBack(dbConnection);
+            String msg = "Error occurred while adding group : " + groupName;
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new UserStoreException(msg, e);
+        } catch (Exception e) {
+            String errorMessage = "Error occurred while getting database type from DB connection";
+            if (log.isDebugEnabled()) {
+                log.debug(errorMessage, e);
+            }
+            if (e instanceof UserStoreException && ERROR_CODE_DUPLICATE_WHILE_WRITING_TO_DATABASE.getCode()
+                    .equals(((UserStoreException) e).getErrorCode())) {
+                // Duplicate entry
```

Comment:
Missing `.` at the end

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/3847/commits/3bb98bcb742cbac3c64ba202dbbda861a1911280